presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/cache/CachingHiveMetastore.java

```diff
@@ -324,27 +317,31 @@ private PartitionStatistics loadPartitionColumnStatistics(WithIdentity<HiveParti
                 partition.getKey().getHiveTableName().getDatabaseName(),
                 partition.getKey().getHiveTableName().getTableName(),
                 ImmutableSet.of(partitionName));
-        if (!partitionStatistics.containsKey(partitionName)) {
+        PartitionStatistics result = partitionStatistics.get(partitionName);
+        if (result == null) {
             throw new PrestoException(HIVE_PARTITION_DROPPED_DURING_QUERY, "Statistics result does not contain entry for partition: " + partition.getKey().getPartitionName());
         }
-        return partitionStatistics.get(partitionName);
+        return result;
     }
 
     private Map<WithIdentity<HivePartitionName>, PartitionStatistics> loadPartitionColumnStatistics(Iterable<? extends WithIdentity<HivePartitionName>> keys)
     {
         SetMultimap<WithIdentity<HiveTableName>, WithIdentity<HivePartitionName>> tablePartitions = stream(keys)
-                .collect(toImmutableSetMultimap(value -> new WithIdentity<>(value.getIdentity(), value.getKey().getHiveTableName()), key -> key));
+                .collect(toImmutableSetMultimap(value -> new WithIdentity<>(value.getIdentity(), value.getKey().getHiveTableName()), Function.identity()));
         ImmutableMap.Builder<WithIdentity<HivePartitionName>, PartitionStatistics> result = ImmutableMap.builder();
         tablePartitions.keySet().forEach(table -> {
-            Set<String> partitionNames = tablePartitions.get(table).stream()
+            Set<WithIdentity<HivePartitionName>> partitionNames = tablePartitions.get(table);
+            ImmutableSet<String> partitionNameStrings = partitionNames.stream()
```

Comment:
nit: use `Set` instead of `ImmutableSet`... we favor the interface name over implementation for collection classes